app/controllers/donations_controller.rb

```diff
@@ -0,0 +1,53 @@
+class DonationsController < ApplicationController
+  skip_before_action :authenticate_user!
+
+  def index
+  end
+
+  def create
+    token = params[:stripeToken]
+    amount = params[:amount].to_i * 100
+    email = if params[:anonymous] == 'off'
+              if user_signed_in?
+                params[:email].presence || current_user.email
+              else
+                params[:email]
+              end
+            end
+
+    customer = Stripe::Customer.create(
+      email: email,
+      source: token
+    )
+
+    stripe_charge = Stripe::Charge.create(
+      amount: amount,
+      currency: "usd",
+      customer: customer.id
+    )
+
+    charge = Charge.create(
+      amount: amount / 100,
+      email: email,
+      description: stripe_charge.description,
+      uid: stripe_charge.id,
+      customer_id: customer.id,
+      charge_type: 'donation'
+    )
+
+    if user_signed_in? && params[:anonymous] == 'off'
+      charge.update(user_id: current_user.id)
+    end
+
```

Comment:
Send an email with the charge uid - if they accidentally did not include their email but did not mean to make an anonymous donation